From 89fa2f158aaa4c3b0eae51be35f40ffd30bfaab4 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Fri, 29 Apr 2022 14:37:24 +0000
Subject: [PATCH 19/28] systemd: Remove systemd-run domain.

This command should be run with the privs of the caller.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/kernel/corecommands.te | 2 +-
 policy/modules/system/systemd.fc      | 1 -
 policy/modules/system/systemd.te      | 4 ----
 3 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/policy/modules/kernel/corecommands.te b/policy/modules/kernel/corecommands.te
index 447cf17eb..dabd3b066 100644
--- a/policy/modules/kernel/corecommands.te
+++ b/policy/modules/kernel/corecommands.te
@@ -14,7 +14,7 @@ attribute exec_type;
 # bin_t is the type of files in the system bin/sbin directories.
 #
 type bin_t alias { ls_exec_t sbin_t };
-typealias bin_t alias systemd_detect_virt_t;
+typealias bin_t alias { systemd_detect_virt_t systemd_run_exec_t };
 corecmd_executable_file(bin_t)
 dev_associate(bin_t)	#For /dev/MAKEDEV
 
diff --git a/policy/modules/system/systemd.fc b/policy/modules/system/systemd.fc
index 756224dca..abe902636 100644
--- a/policy/modules/system/systemd.fc
+++ b/policy/modules/system/systemd.fc
@@ -10,7 +10,6 @@
 /usr/bin/systemd-coredump		--	gen_context(system_u:object_r:systemd_coredump_exec_t,s0)
 /usr/bin/systemd-hwdb			--	gen_context(system_u:object_r:systemd_hw_exec_t,s0)
 /usr/bin/systemd-nspawn			--	gen_context(system_u:object_r:systemd_nspawn_exec_t,s0)
-/usr/bin/systemd-run			--	gen_context(system_u:object_r:systemd_run_exec_t,s0)
 /usr/bin/systemd-stdio-bridge		--	gen_context(system_u:object_r:systemd_stdio_bridge_exec_t,s0)
 /usr/bin/systemd-sysusers		--	gen_context(system_u:object_r:systemd_sysusers_exec_t,s0)
 /usr/bin/systemd-tmpfiles		--	gen_context(system_u:object_r:systemd_tmpfiles_exec_t,s0)
diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index c2e841358..d3c2c3802 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -224,10 +224,6 @@ init_system_domain(systemd_resolved_t, systemd_resolved_exec_t)
 type systemd_resolved_runtime_t alias systemd_resolved_var_run_t;
 files_runtime_file(systemd_resolved_runtime_t)
 
-type systemd_run_t;
-type systemd_run_exec_t;
-init_daemon_domain(systemd_run_t, systemd_run_exec_t)
-
 type systemd_stdio_bridge_t;
 type systemd_stdio_bridge_exec_t;
 init_system_domain(systemd_stdio_bridge_t, systemd_stdio_bridge_exec_t)
-- 
2.25.1

